exonum/src/storage/keys.rs

```diff
@@ -70,131 +74,131 @@ pub trait StorageKey {
 
     /// Deserialize a key from the specified buffer of bytes.
     // TODO: should be unsafe (ECR-174)?
-    fn read(buffer: &[u8]) -> Self;
+    fn read(buffer: &[u8]) -> Self::Owned;
 }
 
 impl StorageKey for () {
     fn size(&self) -> usize {
-        0
+        mem::size_of::<Self>()
```

Comment:
It looks like the right change (we get rid of the magic constants), but in fact it is logically incorrect. The function `mem::size_of` is depends on architecture. For example, you can imagine some strange architecture on which u16 always takes 3 bytes (why not?).  Hovewer, `BigEndian::write_u16(buffer, *self)` always writes 2 bytes, regardless of architecture.

Since the amount of memory used by different types during serialization is part of the *agreement of the protocol*, it is correct to use constants rather than information about how many bytes these types occupy in memory on a given architecture.

Reply:
@defuz https://doc.rust-lang.org/std/mem/fn.size_of.html
>In general, the size of a type is not stable across compilations, but specific types such as primitives are.

Reply:
@vldm Interesting. 👍  There is even a nice table by the link. You think that `size_of` usage in this case is valid, right?

Reply:
@DarkEld3r nope :) This link was just for info about rust.
I am pretty sure, that size of value is highly related with `write` \ `read` implementation, so it should be known and probably finite.